src/shared/components/query/DataTypePrioritySelector.tsx

```diff
@@ -20,33 +24,84 @@ export default class DataTypePrioritySelector extends QueryStoreComponent<{}, {}
 
+		let flexRowContents:JSX.Element[] = [];
+		flexRowContents.push(<LoadingIndicator key="loading" isLoading={this.profileAvailability.isPending}/>);
+		if (this.profileAvailability.isError) {
+			flexRowContents.push(<span key="error">Error loading profiles for selected studies.</span>);
+		} else if (this.profileAvailability.isComplete) {
+			flexRowContents = flexRowContents.concat(radioButtons(this.profileAvailability.result));
+		}
 		return (
 			<FlexRow padded className={styles.DataTypePrioritySelector}>
 				<SectionHeader className="sectionLabel">Select Data Type Priority:</SectionHeader>
-
 				<FlexRow>
-					<this.DataTypePriorityRadio label='Mutation and CNA' state={{mutation: true, cna: true}}/>
-					<this.DataTypePriorityRadio label='Only Mutation' state={{mutation: true, cna: false}}/>
-					<this.DataTypePriorityRadio label='Only CNA' state={{mutation: false, cna: true}}/>
+					{flexRowContents}
 				</FlexRow>
-
 			</FlexRow>
 		);
 	}
 
-	DataTypePriorityRadio = observer(
-		(props: {label: string, state:QueryStore['dataTypePriority']}) => (
-			<label className={styles.DataTypePriorityLabel}>
-				<input
-					type="radio"
-					checked={_.isEqual(toJS(this.store.dataTypePriority), props.state)}
-					onChange={event => {
-						if (event.currentTarget.checked)
-							this.store.dataTypePriority = props.state;
-					}}
-				/>
-				{props.label}
-			</label>
-		)
-	);
+	readonly profileAvailability = remoteData({
+		await:()=>[this.store.molecularProfilesInSelectedStudies],
+		invoke:()=>{
+			return Promise.resolve(profileAvailability(this.store.molecularProfilesInSelectedStudies.result!));
+		}
+	});
+}
+
+export const DataTypePriorityRadio = observer(
+	(props: {label: string, state:QueryStore['dataTypePriority']}) => (
+		<label className={styles.DataTypePriorityLabel}>
+			<input
+				type="radio"
+				checked={_.isEqual(toJS(this.store.dataTypePriority), props.state)}
+				onChange={event => {
+					if (event.currentTarget.checked)
+						this.store.dataTypePriority = props.state;
+				}}
+			/>
+			{props.label}
+		</label>
+	)
+);
+
+export function radioButtons(availability:{mutation:boolean, cna:boolean}):JSX.Element[] {
+	let buttons = [];
+	let hasBoth = false;
+	if (availability.mutation && availability.cna) {
+		buttons.push(<DataTypePriorityRadio key="MC" label='Mutation and CNA' state={{mutation: true, cna: true}}/>);
+		hasBoth = true;
+	}
+	if (availability.mutation) {
+		buttons.push(<DataTypePriorityRadio key="M" label={`${hasBoth ? 'Only ' : ''}Mutation`} state={{mutation:true, cna:false}}/>);
+	}
+	if (availability.cna) {
+		buttons.push(<DataTypePriorityRadio key="C" label={`${hasBoth ? 'Only ' : ''}CNA`} state={{mutation:false, cna:true}}/>);
+	}
+	return buttons;
+}
+
+export function profileAvailability(molecularProfiles:MolecularProfile[]) {
+	let hasMutationProfile = false;
+	let hasCNAProfile = false;
+	for (const profile of molecularProfiles) {
+		if (!profile.showProfileInAnalysisTab)

```

Comment:
just curious but what is reason a profile would have this set to false?

Reply:
showProfileInAnalysisTab

Reply:
im not sure but it is a parameter